samples/bluetooth/alexa_gadget/CMakeLists.txt

```diff
@@ -13,7 +13,7 @@ target_sources(app PRIVATE
   src/main.c
 )
 
-add_subdirectory(src/gadgets)
-target_link_libraries(app PRIVATE ALEXA_GADGETS_PROFILE)
+# Explicit linking needed to ensure protobuffer files are compiled before sample
+target_link_libraries(app PRIVATE BT_ALEXA_GADGETS_PROFILE)
```

Comment:
This should be moved to the gadgets_profile/Cmakelists.txt, otherwise all users of BT_GADGETS will have to replicate this.

It looks a bit suspicious though. Why do they are have to be compiled first?
@tejlmand Do you have any input here?

Reply:
just tried removing it.
It seems that the generated headers for BT_ALEXA_GADGETS_PROFILE is needed when compiling the `app` lib.
And if there is no dependency (either direct or indirect) from app to BT_ALEXA_GADGETS_PROFILE, then it might try to compile the sources for app lib before header files are generated.

This should be improved, when Alexa sources are now being moved under `subsys/bluetooth`

Reply:
Removed this from app cmakelists, and added APP_LINK_WITH_ALEXA_GADGETS_PROTO kconfig option instead